components/lib/multiselect/MultiSelect.js

```diff
@@ -653,6 +657,10 @@ export const MultiSelect = React.memo(
                 return ObjectUtils.isFunction(props.optionDisabled) ? props.optionDisabled(option) : ObjectUtils.resolveFieldData(option, props.optionDisabled);
             }
 
+            if (props.selectionLimit && props.value && props.value.length >= props.selectionLimit && !isSelected(option)) {
```

Comment:
I don't understand why this check is needed. If the user exceeds the selected limit, other options do not need to be disabled.

Reply:
Hi @mertsincan 
So how should `selectionLimit` work? Can users select more than the selection limit?

Reply:
according to prime react JSDoc selection limit should limit the amount of selections. 

# This is obviously a broken prop

Reply:
OK I made some tweaks and thoroughly tested this.  It is in line with how PrimeNG and PrimeVue handle this scenario.   @mertsincan the reason we need to check if the current option is already selected is to allow them to unselect it.  So in the scenario of `selectionLimit={1}` they need to be able to unselect the selected one

![image](https://github.com/primefaces/primereact/assets/4399574/3ef81b57-932a-41ce-8395-7da9a923c671)